linkerd/proxy/spire-client/src/lib.rs

```diff
@@ -0,0 +1,284 @@
+#![deny(rust_2018_idioms, clippy::disallowed_methods, clippy::disallowed_types)]
+#![forbid(unsafe_code)]
+
+mod api;
+
+pub use api::{Api, SvidUpdate};
+use linkerd_error::{Error, Result};
+use linkerd_identity::Credentials;
+use linkerd_identity::Id;
+use std::fmt::Display;
+use std::time::{Duration, UNIX_EPOCH};
+use tokio::sync::watch;
+use tower::Service;
+use tracing::error;
+
+pub struct Spire {
+    id: Id,
+}
+
+// === impl Spire ===
+
+impl Spire {
+    pub fn new(id: Id) -> Self {
+        Self { id }
+    }
+
+    pub async fn run<C, S>(self, credentials: C, mut client: S)
```

Comment:
How should run handle errors?  Our options are basically (1) panic or (2) bubble them up. Given the choice, it's _gnerally_ preferable to bubble them up and, if necessary, panic closer to the root of the callstack.